static/src/js/pages/KeywordManagerPage/KeywordManagerPage.jsx

```diff
@@ -300,6 +346,58 @@ const KeywordManagerPage = () => {
         message="You are now viewing the live published keyword version. Changes made to this version will show up on the website right away."
         actions={warningModalActions}
       />
+      <CustomModal
+        show={showPublishModal}
+        toggleModal={() => setShowPublishModal(false)}
+        header="Publish New Keyword Version"
+        message={
+          (
+            <>
+              <Form.Group>
+                <Form.Label htmlFor="newKeywordVersion">Version Name:</Form.Label>
+                <Form.Control
+                  id="newKeywordVersion"
+                  type="text"
+                  value={newVersionName}
+                  onChange={(e) => setNewVersionName(e.target.value)}
+                  placeholder="Enter version"
+                />
+              </Form.Group>
+              {publishError && <div className="text-danger mt-2">{publishError}</div>}
+            </>
+          )
+        }
+        actions={
+          [
+            {
+              label: 'Cancel',
+              variant: 'secondary',
+              onClick: () => setShowPublishModal(false)
+            },
+            {
+              label: 'Publish',
+              variant: 'primary',
+              onClick: handlePublishVersion
+            }
+          ]
+        }
+      />
+      <CustomModal
+        show={publishSuccess}
+        toggleModal={() => setPublishSuccess(false)}
```

Comment:
```suggestion
        toggleModal={setPublishSuccess(false)}
```

Reply:
call setPublishSuccess directly calls redrawing in an infinite loop

Reply:
Interesting okay disregard this comment thanks